### PR TITLE
snapping for imguizmo

### PR DESCRIPTION
**Pull request by tijon1**
no idea if it works in the editor and assetviewer, but it works fine in a project of mine that i tested it on.

### CODE REVIEW

**Comment by gleblebedev** [comment]
I guess the PR is missing editor changes but it could be done in a separate PR, I guess.

**Comment by gleblebedev** [comment]
What do you think of this option? https://github.com/rbfx/rbfx/pull/396

File: Source/Urho3D/SystemUI/Gizmo.cpp [hunk 1]
```diff
@@ -137,6 +137,7 @@ bool Gizmo::Manipulate(const Camera* camera, Node** begin, Node** end)
     Matrix4 proj = camera->GetProjection().Transpose();
     Matrix4 tran = currentOrigin.Transpose();
     Matrix4 delta;
+    float transSnap[3] = { *snap, *snap, *snap };
```
[the remark this answers is not in the document]
Comment (tijon1) [reply Source/Urho3D/SystemUI/Gizmo.cpp:140]: there's not a need for nullptr check as you would pass in a nullptr anyways if there wasn't any snap

**Comment by tijon1** [comment]
> What do you think of this option? #396

looks good

**Comment by eugeneko** [comment]
Should be fixed on Editor merge, reopen PR/issue if not.

**Pull request closed without merging**